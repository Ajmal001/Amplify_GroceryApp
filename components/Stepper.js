import React from "react";
import { View, TouchableOpacity, StyleSheet, Platform } from "react-native";
import { MaterialCommunityIcons } from "@expo/vector-icons";
import { blue, white, grey, mainColor, secondaryColor } from "../utils/helpers";
import PropTypes from "prop-types";

export default function Stepper({ onIncrement, onDecrement }) {
  return (
    <View style={[styles.row, { justifyContent: "space-between" }]}>
        <View style={{ flexDirection: "row" }}>
          <TouchableOpacity style={styles.androidBtn} onPress={onDecrement}>
            <MaterialCommunityIcons name="minus" size={30} color={secondaryColor} />
          </TouchableOpacity>
          <TouchableOpacity style={styles.androidBtn} onPress={onIncrement}>
            <MaterialCommunityIcons name="plus" size={30} color={secondaryColor} />
          </TouchableOpacity>
        </View>
    </View>
  );
}

const styles = StyleSheet.create({
  row: {
    flex: 1,
    flexDirection: "row",
    alignItems: "center",
  },
  androidBtn: {
    margin: 5,
    backgroundColor: mainColor,
    padding: 10,
    borderRadius: 10,
  },
});

Stepper.propTypes = {
  onIncrement: PropTypes.func.isRequired,
  onDecrement: PropTypes.func.isRequired,
};
