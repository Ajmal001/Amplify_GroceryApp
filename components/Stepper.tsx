import React from "react";
import { View, TouchableOpacity, StyleSheet } from "react-native";
import { MaterialCommunityIcons } from "@expo/vector-icons";
import { mainColor, secondaryColor } from "../utils/helpers";
import PropTypes from "prop-types";

export interface Props {
  onIncrement: () => void;
  onDecrement: () => void;
}



export const Stepper: React.FC<Props> = ({ onIncrement, onDecrement }) => {
  return (
    <View style={[styles.row, { justifyContent: "space-between" }]}>
      <View style={{ flexDirection: "row" }}>
        <TouchableOpacity style={styles.androidBtn} onPress={onDecrement}>
          <MaterialCommunityIcons name="minus" size={30} color={secondaryColor} />
        </TouchableOpacity>
        <TouchableOpacity style={styles.androidBtn} onPress={onIncrement}>
          <MaterialCommunityIcons name="plus" size={30} color={secondaryColor} />
        </TouchableOpacity>
      </View>
    </View>
  );
}

export default Stepper;

const styles = StyleSheet.create({
  row: {
    flex: 1,
    flexDirection: "row",
    alignItems: "center",
  },
  androidBtn: {
    margin: 5,
    backgroundColor: mainColor,
    padding: 10,
    borderRadius: 10,
  },
});

Stepper.propTypes = {
  onIncrement: PropTypes.func.isRequired,
  onDecrement: PropTypes.func.isRequired,
};
