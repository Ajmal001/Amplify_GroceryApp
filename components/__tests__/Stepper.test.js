import React from 'react';
import renderer from 'react-test-renderer';
import Stepper from '../Stepper';

jest.mock("@expo/vector-icons", () => {
    return {
      MaterialCommunityIcons: () => {
        return <></>;
      },
    };
  });


test('renders correctly', () => {
  const tree = renderer.create(<Stepper
    onIncrement={(x)=>x} 
    onDecrement={(x)=>x}/>).toJSON();
  expect(tree).toMatchSnapshot();
});