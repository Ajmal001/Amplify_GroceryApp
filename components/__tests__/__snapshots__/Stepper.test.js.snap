// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      },
      Object {
        "justifyContent": "space-between",
      },
    ]
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#ffca18",
          "borderRadius": 10,
          "margin": 5,
          "opacity": 1,
          "padding": 10,
        }
      }
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#ffca18",
          "borderRadius": 10,
          "margin": 5,
          "opacity": 1,
          "padding": 10,
        }
      }
    />
  </View>
</View>
`;
